### PR TITLE
Async command update

### PR DESCRIPTION
**Pull request by slluis**
Adds support for async command update handlers. If an update handler returns a task, changes done in the provided CommandUpdaterInfo will be shown in the corresponding menu item.

```
[CommandUpdateHandler (ViewCommands.OpenWithList)]
public async Task OnOpenWithUpdate (CommandArrayInfo info, CancellationToken ct)
{
    info.Add (new CommandInfo ("Loading...") { Enabled = false }, null);
    var pf = (ProjectFile)CurrentNode.DataItem;
    // Some potentially slow operation
    var viewers = await DisplayBindingService.GetFileViewers (pf.FilePath, pf.Project, ct);
    info.Clear ();
    PopulateOpenWithViewers (info, viewers);
}
```

The PR also removes menu creation infrastructure from the Mac add-in, since we now have the same code in MD.Ide.

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandInfo.cs [hunk 1]
```diff
@@ -77,47 +82,47 @@ public CommandInfo (string text, bool enabled, bool checkd)
 		
 		public string Text {
 			get { return text; }
-			set { text = value; }
+			set { text = value; NotifyChanged (); }
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandInfo.cs]: Wouldn't we benefit more from IPropertyChanged here? We'd also have the same simplified code if we used `[CallerMemberName]` with the added benefit of being able to filter which value changed.

Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandInfo.cs]: I don't think it is worth the additional complexity.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandManager.cs [hunk 2]
```diff
@@ -486,6 +488,8 @@ internal bool ProcessKeyEvent (Gdk.EventKey ev)
 
 			for (int i = 0; i < commands.Count; i++) {
 				CommandInfo cinfo = GetCommandInfo (commands[i].Id, new CommandTargetRoute ());
+				if (cinfo.IsUpdatingAsynchronously)
+					cinfo.UpdateTask.Wait (); // Not nice, but we need a synchronous result here
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.Commands/CommandManager.cs]: Maybe it's time to look into async-ifying keypresses and handling keypresses in order?

**Comment by slluis** [comment]
Replaced by #1904

**Pull request closed without merging**